pkg/appmanager/appManager.go

```diff
@@ -90,20 +89,30 @@ type Manager struct {
 	broadcaster   record.EventBroadcaster
 	eventRecorder record.EventRecorder
 	eventSource   v1.EventSource
+	// Route configurations
+	routeConfig RouteConfig
 }
 
 // Struct to allow NewManager to receive all or only specific parameters.
 type Params struct {
 	KubeClient      kubernetes.Interface
 	restClient      rest.Interface // package local for unit testing only
-	RouteClientV1   *routeclient.Client
+	RouteClientV1   rest.Interface
```

Comment:
Right now we use the KubeClient to send updates back to k8s for Ingress/ConfigMaps. Using this rest.Interface allows the unit tests to be able to mock it properly. Maybe there is a way to also pull in routeclient.Client as a separate var for updating Routes.